### ISSUE TITLE
support for "real" toggle switches?

### ISSUE THREAD

**Issue by ale-79** [stella-emu/stella#329]
What about adding support for physical toggle switches for TV_TYPE and LEFT/RIGHT DIFFICULTY?
That would be exactly the same as what the current "7800 pause key" (#245) does, but extended to the difficulty switches as well. 

I think this could be an interesting feature for those building a dedicated console running Stella on a raspberry PI, (or modding a retron77), but also PC users might like to have a control panel that more closely mimics the original hardware. There are a lot of cheap USB encoders available (typically used for building arcade sticks and control panels for MAME) that can be used, or an existing controller can be modded: it's just a matter of installing toggle switches instead of momentary push buttons.

**Comment by thrust26** [comment]
Have you checked the Input dialog? You can already map keys to toggle those. 

Or do I misunderstand you here?

**Comment by ale-79** [comment]
I know you can map a (momentary, as in every PC keyboard or controller) key to toggle the emulated switches. 
What I mean is to have a custom built controller using real toggle switches instead of momentary buttons. Stella doesn't have to toggle the emulated switch in that case, but just read the status of the real switch

**Comment by thrust26** [comment]
As soon as there is hardware there will be software. :smile:

**Comment by sa666666** [comment]
@ale-79, did anything ever come of this?  As @thrust26 said, we can't do any software changes unless/until we have hardware to test.  Else we can close this one.

**Comment by ale-79** [comment]
Nope, I had this idea of building a custom arcade usb controller that included the console switches, but I actually never did.
We can close this.
